Core/Core/Login/LoginWebPresenter.swift

```diff
@@ -109,9 +109,10 @@ class LoginWebPresenter {
             return .cancel
         }
 
-        // I dunno why, but we have to wait for the code to be the first param cuz it can keep changing as we follow redirects
-        //  "/canvas/login?code="
-        if let code = codeFromQueryItems(queryItems), let mobileVerify = mobileVerifyModel, let baseURL = mobileVerify.base_url {
+        if // wait for "https://canvas/login?code="
+            url.absoluteString.hasPrefix("https://canvas/login"),
+            let code = queryItems?.first(where: { $0.name == "code" })?.value, !code.isEmpty,
+            let mobileVerify = mobileVerifyModel, let baseURL = mobileVerify.base_url {
```

Comment:
A more robust check than "is code the first query item?"

this is a weird url though, so `components.host` and `components.path` are both nil, which is why we check `absoluteString`.